### DISCUSSION TITLE
effect of language parameter

### DISCUSSION THREAD

**Discussion by xinyuluo** [komoot/photon#682] [Q&A]
Hi! I was under the impression that the language parameter only affects the value returned in the response (e.g. `Aragón` vs `Aragon`) but not the list of objects. Yet when I tried the two following queries, results are different:
```
$ curl 'localhost:2322/api?q=capella&lat=1.35209983107&lon=103.82&limit=1'
{"features":[{"geometry":{"coordinates":[101.4471801,0.5243353],"type":"Point"},"type":"Feature","properties":{"osm_id":6959621248,"country":"Indonesia","city":"Pekanbaru","countrycode":"ID","postcode":"28144","county":"Kabupaten Kampar","type":"house","osm_type":"N","osm_key":"shop","street":"Jalan Jenderal Sudirman","osm_value":"yes","name":"Capella Honda Sudirman","state":"Riau"}}],"type":"FeatureCollection"}

$ curl 'localhost:2322/api?q=capella&lat=1.35209983107&lon=103.82&limit=1&lang=en'
{"features":[{"geometry":{"coordinates":[103.8244312,1.2496982],"type":"Point"},"type":"Feature","properties":{"osm_id":4302905927,"country":"Singapore","city":"Singapore","countrycode":"SG","postcode":"098297","county":"Southwest","type":"house","osm_type":"N","osm_key":"tourism","housenumber":"1","street":"The Knolls","district":"Southern Islands","extra":{"phone":"+65 6377 8888","website":"https://www.capellahotels.com/singapore/"},"osm_value":"hotel","name":"Capella Singapore"}}],"type":"FeatureCollection"}
```
Could someone please help me understand why `lang` plays a role in the result change here? Does photon have totally different/separate search index based on the language choice? 
Thank you 🙏

**Comment by lonvia** [comment]
The user language is used to rescore the results. There are various places in the query builder where the default language means that a different weight factor is used. For example in this line https://github.com/komoot/photon/blob/2395ab4543851026ba83ac9e2acc76d26fa57bd6/src/main/java/de/komoot/photon/elasticsearch/PhotonQueryBuilder.java#L78
